ui/shared/chart/ChartMenu.tsx

```diff
@@ -0,0 +1,181 @@
+              'transform-origin': 'top left',
+            },
+          })
+          .then((dataUrl) => {
+            const link = document.createElement('a');
+            link.download = `${ title } (Blockscout chart).png`;
+            link.href = dataUrl;
+            link.click();
+            link.remove();
+          });
+      }
+    }, 100);
+  }, [ pngBackgroundColor, title, chartRef ]);
+
+  const handleSVGSavingClick = React.useCallback(() => {
+    if (items) {
+      const headerRows = [
+        'Date', 'Value',
+      ];
+      const dataRows = items.map((item) => [
+        dayjs(item.date).format('YYYY-MM-DD'), String(item.value),
+      ]);
+
+      saveAsCSV(headerRows, dataRows, `${ title } (Blockscout stats)`);
+    }
+  }, [ items, title ]);
+
+  // TS thinks window.navigator.share can't be undefined, but it can
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const hasShare = isInBrowser && (window.navigator.share as any);
+
+  const handleShare = React.useCallback(async() => {
+    try {
+      await window.navigator.share({
+        title: title,
+        text: description,
+        url: chartUrl,
+      });
+    } catch (error) {}
+  }, [ title, description, chartUrl ]);
+
+  return (
+    <>
+      <Menu>
+        <Skeleton isLoaded={ !isLoading } borderRadius="base">
+          <MenuButton
+            w="36px"
+            h="32px"
+            icon={ <IconSvg name="dots" boxSize={ 4 } transform="rotate(-90deg)"/> }
+            colorScheme="gray"
+            variant="simple"
+            as={ IconButton }
+          >
+            <VisuallyHidden>
+            Open chart options menu
+            </VisuallyHidden>
+          </MenuButton>
+        </Skeleton>
+        <MenuList>
+          { href && (
+            <MenuItem
+              display="flex"
+              alignItems="center"
+              onClick={ hasShare ? handleShare : onCopy }
+              closeOnSelect={ hasShare ? false : true }
+            >
+              <IconSvg name={ hasShare ? 'share' : 'copy' } boxSize={ 5 } mr={ 3 }/>
+              { hasShare ? 'Share' : 'Copy link' }
+            </MenuItem>
+          ) }
+          <MenuItem
+            display="flex"
+            alignItems="center"
+            onClick={ showChartFullscreen }
+          >
+            <IconSvg name="scope" boxSize={ 5 } mr={ 3 }/>
+            View fullscreen
+          </MenuItem>
+          <MenuItem
+            display="flex"
+            alignItems="center"
+            onClick={ handleFileSaveClick }
+          >
+            <IconSvg name="files/image" boxSize={ 5 } mr={ 3 }/>
+            Save as PNG
+          </MenuItem>
+          <MenuItem
+            display="flex"
+            alignItems="center"
+            onClick={ handleSVGSavingClick }
+          >
+            <IconSvg name="files/csv" boxSize={ 5 } mr={ 3 }/>
+            Save as CSV
+          </MenuItem>
+        </MenuList>
+      </Menu>
+      { items && (
```

Comment:
I suppose we can render the modal only if it is opened
```suggestion
      { items && isFullscreen && (
```